File: crates/cxx-swift/cxx/lib/swift/AST/ASTWalkerBase.cxx
#include "cxx-swift-auto/cxx/include/swift/AST/ASTWalkerBase.hxx"

namespace cxx_swift::swift::ast::ast_walker_base {

[[gnu::always_inline]]
auto
ASTWalkerBase::shouldWalkIntoGenericParams() noexcept -> bool
{
  return this->ASTWalker::shouldWalkIntoGenericParams();
}

[[gnu::always_inline]]
auto
ASTWalkerBase::shouldWalkIntoTapExpression() noexcept -> bool
{
  return this->ASTWalker::shouldWalkIntoTapExpression();
}

[[gnu::always_inline]]
auto
ASTWalkerBase::shouldWalkIntoPropertyWrapperPlaceholderValue() noexcept -> bool
{
  return this->ASTWalker::shouldWalkIntoPropertyWrapperPlaceholderValue();
}

[[gnu::always_inline]]
auto
ASTWalkerBase::shouldWalkCaptureInitializerExpressions() noexcept -> bool
{
  return this->ASTWalker::shouldWalkCaptureInitializerExpressions();
}

[[gnu::always_inline]]
auto
ASTWalkerBase::shouldWalkAccessorsTheOldWay() noexcept -> bool
{
  return this->ASTWalker::shouldWalkAccessorsTheOldWay();
}

[[gnu::always_inline]]
auto
ASTWalkerBase::shouldWalkSerializedTopLevelInternalDecls() noexcept -> bool
{
  return this->ASTWalker::shouldWalkSerializedTopLevelInternalDecls();
}

} // namespace cxx_swift::swift::ast::ast_walker_base

File: crates/cxx-swift/src/ffi/swift/ast/ast_walker_base.rs
use crate::{ffi::swift::ast::ast_walker::AstWalker, gen::swift::ast::ast_walker_base};
use core::pin::Pin;

pub use crate::auto::swift::ast::ast_walker_base::AstWalkerBase;

impl<'ctx> core::ops::Deref for AstWalkerBase<'ctx> {
    type Target = AstWalker<'ctx>;

    #[inline]
    fn deref(&self) -> &Self::Target {
        ast_walker_base::deref(self)
    }
}

impl<'ctx> AstWalkerBase<'ctx> {
    #[inline]
    pub fn deref_pin(self: Pin<&mut Self>) -> Pin<&mut AstWalker<'ctx>> {
        ast_walker_base::deref_pin(self)
    }
}

impl<'ctx> AstWalkerBase<'ctx> {
    #[inline]
    pub unsafe fn new() -> impl moveref::New<Output = AstWalkerBase<'ctx>> {
        Self::default_new()
    }
}

impl<'ctx> AstWalkerBase<'ctx> {
    #[inline]
    pub unsafe fn should_walk_into_generic_params(self: Pin<&mut Self>) -> bool {
        ast_walker_base::should_walk_into_generic_params(self)
    }

    #[inline]
    pub unsafe fn should_walk_into_tap_expression(self: Pin<&mut Self>) -> bool {
        ast_walker_base::should_walk_into_tap_expression(self)
    }

    #[inline]
    pub unsafe fn should_walk_into_property_wrapper_placeholder_value(self: Pin<&mut Self>) -> bool {
        ast_walker_base::should_walk_into_property_wrapper_placeholder_value(self)
    }

    #[inline]
    pub unsafe fn should_walk_capture_initializer_expressions(self: Pin<&mut Self>) -> bool {
        ast_walker_base::should_walk_capture_initializer_expressions(self)
    }

    #[inline]
    pub unsafe fn should_walk_accessors_the_old_way(self: Pin<&mut Self>) -> bool {
        ast_walker_base::should_walk_accessors_the_old_way(self)
    }

    #[inline]
    pub unsafe fn should_walk_serialized_top_level_internal_decls(self: Pin<&mut Self>) -> bool {
        ast_walker_base::should_walk_serialized_top_level_internal_decls(self)
    }
}
